contracts/lupi.sol

```diff
@@ -38,12 +61,13 @@ contract lupi is owned {
     }
 
     function placeBet(bytes32 _secretBet) payable returns (uint ticket) {
-        require (state == State.Betting);
+        require (state == State.Betting && msg.value == requiredBetAmount
+            && ticketCountLimit < tickets.length -1);
```

Comment:
It's a better pattern to make additional preconditions a new require(...) line instead of &&. Will support extracting to a modifier later, and helps with debugging too (esp. if they ever add line number reporting on `throw` :)